hdr_approx_percentile.sql

```diff
@@ -0,0 +1,232 @@
+-- --------
+-- Overview
+-- --------
+--
+-- This is a demo test to show how to use hdr_histogram (http://hdrhistogram.org/)
+-- to compute `approx_percentile` in the RisingWave stream engine.
+--
+-- The advantage of this approach is that it supports updatable approx_percentile.
+-- The disadvantage is that it requires quadratic computation and storage
+-- as the precision increases,
+-- due to streaming nested loop join.
+--
+-- The way it works is that hdr_histogram stores N significant digits of each value,
+-- thereby reducing the number of buckets needed to store the distribution.
+--
+-- ----------------------
+-- Implementation details
+-- ----------------------
+--
+-- First we create the histogram itself, where each value is encoded as a triple:
+-- 1. The sign of the value (1 or -1)
+-- 2. The exponent of the value (the power of 10, truncated to an integer)
+-- 3. The significand of the value (the digits after the decimal point, truncated to $PRECISION digits)
+--
+-- With the following parameters:
+-- precision=2 (significand is 2 digits)
+-- value=123456
+--
+-- The exponent will be 5 (log10(123456) = 5.0915... ~ 5)
+-- The significand will be 23 (123456 / 10^5 - 1 = 1.23... ).
+-- Then the histogram will store the triple (1, 5, 23)
+--
+-- Next we do a window aggregation on the histogram to compute the
+-- cumulative frequency of each bucket.
```

Comment:
adding the definition of how we compute buckets would be helpful

in this case, (1 + 23/10^2)*(10^5) = 12300